File: frontend/src/services/dashboard.service.ts
import axiosInstance from '../config/axios.instance';
import { ApiResponse } from '../types/api';

export interface DashboardStatisticsResponse {
  totalStudents: number;
  totalClasses: number;
  totalTeachers: number;
}

export const dashboardService = {
  getStatistics: async (): Promise<ApiResponse<DashboardStatisticsResponse>> => {
    const response = await axiosInstance.get('/api/core/dashboard/statistics');
    return response.data;
  },
};

File: preschool-management-backend/src/main/java/com/vusystem/preschool_management_backend/modules/core/services/DashboardService.java
package com.vusystem.preschool_management_backend.modules.core.services;

import com.vusystem.preschool_management_backend.modules.core.dto.response.DashboardStatisticsResponse;

public interface DashboardService {
    DashboardStatisticsResponse getDashboardStatistics();
}
